server/service/apple_mdm.go

```diff
@@ -822,6 +823,141 @@ func generateEnrollmentProfileMobileconfig(orgName, fleetURL, scepChallenge, top
+	default:
+		// ok
+	}
+
+	cmdUUID, err := svc.enqueueMDMAppleCommandRemoveEnrollmentProfile(ctx, h.UUID)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "enqueuing mdm apple remove profile command")
+	}
+
+	return svc.pollResultMDMAppleCommandRemoveEnrollmentProfile(ctx, cmdUUID, h.UUID)
+}
+
+func (svc *Service) enqueueMDMAppleCommandRemoveEnrollmentProfile(ctx context.Context, hostUUID string) (string, error) {
+	cmd := new(mdm.Command)
+	cmdUUID := uuid.New().String()
+	cmd.CommandUUID = cmdUUID
+	cmd.Command.RequestType = "RemoveProfile"
+	cmd.Raw = []byte(generateMDMAppleCommandRemoveEnrollmentProfile(cmdUUID, apple_mdm.FleetPayloadIdentifier))
+
+	status, res, err := rawCommandEnqueue(ctx, svc.mdmStorage, svc.mdmPushService, cmd, []string{hostUUID}, false, svc.logger)
+	if err != nil {
+		return cmdUUID, ctxerr.Wrap(ctx, err)
+	}
+
+	// Additional logging to aid in debugging unexpected results
+	if status != http.StatusOK {
+		logs := []interface{}{
+			"msg", fmt.Sprintf("enqueuing mdm apple remove profile command resulted in unexpected status %d", status),
+			"host_uuid", hostUUID,
+			"command_uuid", cmdUUID,
+		}
+		if res != nil {
+			if r, ok := res.Status[hostUUID]; !ok {
+				logs = append(logs, "detail", "none")
+			} else {
+				logs = append(logs, "push_result", r.PushResult, "push_error", r.PushError, "command_error", r.CommandError)
+			}
+		}
+		level.Debug(svc.logger).Log(logs...)
+	}
+
+	return cmdUUID, nil
+}
+
+func generateMDMAppleCommandRemoveEnrollmentProfile(cmdUUID string, profileUUID string) string {
+	return fmt.Sprintf(`
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd"&gt;
+<plist version="1.0">
+<dict>
+	<key>CommandUUID</key>
+	<string>%s</string>
+	<key>Command</key>
+	<dict>
+		<key>RequestType</key>
+		<string>RemoveProfile</string>
+		<key>Identifier</key>
+		<string>%s</string>
+	</dict>
+</dict>
+</plist>`, cmdUUID, profileUUID)
+}
+
+func (svc *Service) pollResultMDMAppleCommandRemoveEnrollmentProfile(ctx context.Context, cmdUUID string, deviceID string) error {
+	ctx, cancelFn := context.WithDeadline(ctx, time.Now().Add(5*time.Second))
+	ticker := time.NewTicker(300 * time.Millisecond) // TODO confirm interval
+	defer func() {
+		ticker.Stop()
+		cancelFn()
+	}()
+
+	for {
+		select {
+		case <-ctx.Done():
+			// time out after 5 seconds
+			return fleet.MDMAppleCommandTimeoutError{}
+		case <-ticker.C:
+			enabled, err := svc.ds.GetNanoMDMEnrollmentStatus(ctx, deviceID)
+			switch {
+			case err != nil:
+				level.Error(svc.logger).Log("err", "get nanomdm enrollment status", "details", err, "id", deviceID, "command_uuid", cmdUUID)
+				return err
+			case !enabled:
+				// success, mdm enrollment is no longer enabled for the device
+				level.Info(svc.logger).Log("msg", "mdm disabled for device", "id", deviceID, "command_uuid", cmdUUID)
+				return nil
+			default:
```

Comment:
Again, tiny nit but I'd remove the unnecessary clause, continuing is what will happen automatically.